src/crimson/common/logclient.cc

```diff
@@ -0,0 +1,360 @@
+#include "crimson/common/logclient.h"
+#include "include/str_map.h"
+#include "messages/MLog.h"
+#include "messages/MLogAck.h"
+#include "messages/MMonGetVersion.h"
+#include "crimson/net/Messenger.h"
+#include "crimson/mon/MonClient.h"
+#include "mon/MonMap.h"
+#include "common/Graylog.h"
+
+#define dout_subsys ceph_subsys_monc
+
+using std::map;
+using std::ostream;
+using std::ostringstream;
+using std::string;
+
+int parse_log_client_options(CephContext *cct,
```

Comment:
A to-do suggestion for a follow-up PR:

There is a general pattern of:
- creating a set of string-maps
- calling parse_log_client_options() with those maps as out params
-  (parse_log...() calls get_conf_str_map_helper() - which actually calls get_str_map() - to initialize each map)
- passing the whole collection of maps to one or two consumer functions.


I'd suggest we modify the order so that parse_log..() will create a named structure containing the maps,
and return an optional to it. Thus, wrapper functions like update_log_config() or Monitor::update_log_clients(), that do
not manipulate the individual maps, will not have to declare and move them around.